### PR TITLE
internal/ethapi: fix incorrect type on empty slice

### LINKED ISSUE #24368: fields["logs"] incorrectly set
isn't this a bug?

internal/ethapi/api.go , line 1660
```
    if receipt.Logs == nil {
        fields["logs"] = [][]*types.Log{}
    }  
```
the fields["logs"] must be of type `[]*types.Log[]`, not `[][]*types.Log` , i.e. it must be single array, not double array:

as in receipt.go:

```
 type Receipt struct {
....
    Logs              []*Log `json:"logs"              gencodec:"required"`
....
}
```

this is in 1.10.15 (released code tarball)

### PR DESCRIPTION
**Pull request by 0x6f736f646f**
Signed-off-by: 0x6f736f646f <blackd0t@protonmail.com>

Resolves #24368

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ type feeHistoryResult struct {
 	GasUsedRatio []float64        `json:"gasUsedRatio"`
 }
 
+// FeeHistory function
```
Comment (s1na) [inline internal/ethapi/api.go:91]: Nice that you added a comment, but it's not a very helpful one

**Review by s1na — COMMENTED** [review]
Looks good, only one nit

**Review by verheesj — APPROVED** [review]

**Review by Zachinquarantine — APPROVED** [review]

File: internal/ethapi/api.go (hunk 1 above)
Comment (holiman) [reply internal/ethapi/api.go:91]: Please remove this or write something sensible

**Review by holiman — APPROVED** [review]
LGTM aside from the stray comment

File: internal/ethapi/api.go (hunk 1 above)
Comment (holiman) [inline internal/ethapi/api.go:91]:
```suggestion
```

Comment (holiman) [reply internal/ethapi/api.go]: nevermind, I fixed it

**Pull request merged**